src/xunit.performance.api/XunitPerformanceHarness.cs

```diff
@@ -71,6 +75,66 @@ public void RunBenchmarks(string assemblyPath)
             s_profiler(assemblyPath, Configuration.RunId, _outputDirectory, runner, collectOutputFilesCallback);
         }
 
+        public void RunStartupBenchmark(string[] args, ProcessStartInfo psi, PreDel pre, MidDel mid, PostDel post)
+        {
+            pre(_performanceTestConfig, psi);
+            int iterations = _performanceTestConfig.iterations;
+            int timeout = _performanceTestConfig.timeout;
+            psi.RedirectStandardOutput = true;
+
+            // Warmup run.
+            using (var p = new Process()) 
+            {
+                p.StartInfo = psi;
+                p.Start();
+                if (p.WaitForExit(timeout) == false) 
+                {
+                    if (p != null)
+                    {
+                        p.Kill();
+                    }
+                    Console.WriteLine("Timeouted!");
+                    return;
+                }
+            }
+
+            for (int i = 0; i < iterations; i++)
+            {
+                using(var p = new Process())
+                {
+                    p.StartInfo = psi;
+                    p.Start();
+                    if (p.WaitForExit(timeout) == false) 
+                    {
+                        if (p != null)
+                        {
+                            p.Kill();
+                        }
+                        Console.WriteLine("Timeouted!");
+                        return;
+                    }
+                    mid(_performanceTestConfig);
+                }
+            }
```

Comment:
Why this is not a single loop with (Iterations + 1) number of iterations? I think it would be cleaner, you can also capture the output, and decide to ignore it when computing statistics and uploading to BenchView.